teps/0090-matrix.md

```diff
@@ -1340,6 +1312,72 @@ spec:
         name: build-push
 ```
 
+### Status
+
+The status of `PipelineRuns` with fanned-out `PipelineTasks` will list all the `TaskRuns` and `Runs` created.
+
+In `PipelineRuns` with full embedded status, the status of the `PipelineRun` would be as such:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: matrixed-pr
+  ...
+spec:
+  ...
+status:
+  completionTime: "2020-05-04T02:19:14Z"
+  conditions:
+    - lastTransitionTime: "2020-05-04T02:19:14Z"
+      message: "Tasks Completed: 4, Skipped: 0"
+      reason: Succeeded
+      status: "True"
+      type: Succeeded
+  startTime: "2020-05-04T02:00:11Z"  
+  taskRuns:
+    matrixed-pr-foo-0:
+      pipelineTaskName: foo
+      status:
+        ...
+    matrixed-pr-foo-1:
+      pipelineTaskName: foo
+      status:
+        ...
+```
+
+In `PipelineRuns` with minimal embedded status, the status of the `PipelineRun` would be as such:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: matrixed-pr
+  ...
+spec:
+  ...
+status:
+  completionTime: "2020-05-04T02:19:14Z"
+  conditions:
+    - lastTransitionTime: "2020-05-04T02:19:14Z"
+      message: "Tasks Completed: 4, Skipped: 0"
+      reason: Succeeded
+      status: "True"
+      type: Succeeded
+  startTime: "2020-05-04T02:00:11Z"
+  childReferences:
+    - apiVersion: tekton.dev/v1beta1
+      kind: TaskRun
+      name: matrixed-pr-foo-0
+      pipelineTaskName: foo
+    - apiVersion: tekton.dev/v1beta1
```

Comment:
Wonder if we'll run into etcd size issues with a large number of matrixed runs

Reply:
we addressed that through concurrency controls in: https://github.com/tektoncd/community/blob/main/teps/0090-matrix.md#concurrency-control -- and the work in https://github.com/tektoncd/community/pull/716 will be helpful in providing a long-term for concurrency controls (need to add the matrix use case in that TEP)

and using the minimal status also helps us here: https://github.com/tektoncd/community/blob/main/teps/0110-decouple-catalog-organization-and-reference.md

Reply:
@abayer updated the status to use the minimal mode only, regardless of the flag

Reply:
`ChildStatusReference` contains `whenExpressions` in addition to `kind`, `name`, and `pipelineTaskName`. The rationale behind including `whenExpressions` was to audit the resolved inputs and values of the expressions. Similarly, we will need such means to provide the combination of matrix for a particular `taskRun` or `run`. This does not have to be implemented in a single phase but its something I would like to keep track of.

Reply:
@pritidesai by "combination of matrix", do you mean the array of parameters that were used?

Reply:
thank you @jerop, yup, the exact combination of parameters used in taskRun execution. I am guessing these values will end up in the taskRun status eventually (as resolved parameters) but will be handy to have them in `pipelineRun` status.

For example:

```
     matrix:
        - name: platform
          value:
          - linux
          - mac
          - windows
        - name: browser
          value:
          - chrome
          - safari
          - firefox
        - name: version
          value:
          - 1.0
          - 2.0
          - 3.0
```

This matrix produces 27 different combinations, it will be handy to know the name of the `taskRun` for (linux, chrome, 2.0) v/s (linux, chrome, 1.0), so on and so forth. 

This doesn't have to be blocker but I am running a pipeline to produce SBOMs for thousands (of course matrix is limited 256) of packages and will be good to know the name of the `taskRun` for a particular package.  Does it make sense?

Reply:
this was actually in the original proposal and we agreed to remove it in the initial version then only add it later if users need it - https://github.com/tektoncd/community/pull/600#discussion_r803000813 - definitely something we should explore doing after the initial release (cc @afrittoli)